src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs

```diff
@@ -232,6 +232,64 @@ private void OnHeaderTemplateChanged(DataTemplate oldHeaderTemplate, DataTemplat
 			UpdateHeaderVisibility();
 		}
 
+		protected override void PrepareContainerForItemOverride(DependencyObject element, object item)
+		{
+			SynchronizeContainerInheritedProperties(element, item);
+
+			base.PrepareContainerForItemOverride(element, item);
+		}
+
+		protected override void ClearContainerForItemOverride(DependencyObject element, object item)
+		{
+			ClearContainerInheritedProperties(element, item);
+
+			base.ClearContainerForItemOverride(element, item);
+		}
+
+		/// <summary>
+		/// Explicitly propagate inherited properties values to the item containers, as they are not direct visual children of the ComboBox.
+		/// </summary>
+		private void SynchronizeContainerInheritedProperties(DependencyObject element, object item)
+		{
+			if (element is ContentControl cc && !IsItemItsOwnContainer(item))
+			{
+				foreach (var comboBoxProperty in DependencyProperty.GetPropertiesForType(this.GetType()))
```

Comment:
This will get properties for user-defined control that inherits `ComboBox`, does this behavior match WinUI? Consider a case where the user-defined control shadows a property of `ComboBox`, should the user-defined property be set or the original property?

Reply:
User-defined properties cannot be inherited, and the instance is not built by the user. The only situation is that if the user changes the item container style, the CombBox value will take precedence. I don't know if that's what WinUI does at this point, but it's worth looking at it.

Currently the PR is in draft for reasons like these, as there may be side effects in tests.